stage2/include/compare

```diff
@@ -0,0 +1,344 @@
+#ifndef _CXXSHIM_COMPARE
```

Comment:
also this commit message (dda47138d6504c6a471c53621e44005b285c98fc) has an excessively long line